File: NetFramework/TouchToolkit/VB/TouchToolkitDemo/GcZoomPages/Help UserControl/BoundaryFeedbackUI.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports C1.Win.TouchToolKit

Namespace C1ZoomPages
    Partial Public Class BoundaryFeedbackUI
        Inherits UserControl
        Public Sub New()
            InitializeComponent()
        End Sub

        Public Property BoundaryFeedbackMode() As BoundaryFeedbackMode
            Get
                Return If(radioButton1.Checked, BoundaryFeedbackMode.Split, BoundaryFeedbackMode.Standard)
            End Get
            Set
                radioButton1.Checked = (Value = BoundaryFeedbackMode.Split)
            End Set
        End Property

        Private Sub radioButton_CheckedChanged(sender As Object, e As EventArgs)
            Me.OnBoundaryFeedbackModeChanged(e)
        End Sub

        Public Event BoundaryFeedbackModeChanged As EventHandler(Of EventArgs)

        Protected Overridable Sub OnBoundaryFeedbackModeChanged(e As EventArgs)
            RaiseEvent BoundaryFeedbackModeChanged(Me, e)
        End Sub
    End Class
End Namespace

File: Next/TouchToolkit/CS/TouchToolkitExplorer/GcZoomPages/Help UserControl/BoundaryFeedbackUI.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using C1.Win.TouchToolKit;

namespace TouchToolkitExplorer.C1ZoomPages
{
    public partial class BoundaryFeedbackUI : UserControl
    {
        public BoundaryFeedbackUI()
        {
            InitializeComponent();
        }

        public BoundaryFeedbackMode BoundaryFeedbackMode
        {
            get
            {
                return radioButton1.Checked ? BoundaryFeedbackMode.Split : BoundaryFeedbackMode.Standard;
            }
            set
            {
                radioButton1.Checked = (value == BoundaryFeedbackMode.Split);
            }
        }

        private void radioButton_CheckedChanged(object sender, EventArgs e)
        {
            this.OnBoundaryFeedbackModeChanged(e);
        }

        public event EventHandler<EventArgs> BoundaryFeedbackModeChanged;

        protected virtual void OnBoundaryFeedbackModeChanged(EventArgs e)
        {
            if (this.BoundaryFeedbackModeChanged != null)
            {
                this.BoundaryFeedbackModeChanged(this, e);
            }
        }   
    }
}
